### PR TITLE
Add Di to redis-client

### PR DESCRIPTION
**Pull request by cricanr**
Trello: https://trello.com/c/m1YjEipo/8-initialize-redisclient-using-di-instead-of-creating-new-object-using-new
I have added Guice DI for RedisClient instead of using new operator. The code will be easier to test this way. I also went along to use application config values rather then hardcoded values in the project.

### CODE REVIEW

File: conf/application.conf [hunk 1]
```diff
@@ -375,10 +375,10 @@ db {
...
 
 rating.stream {
-  baseUri = "http://localhost:3020"
+  baseUrl = "http://localhost:3020"
   method = "ratings"
 }
 
 redis {
-  ratingsSortedList = "ratingsAvg2"
+  ratingsSortedList = "ratingsSortedList"
```
Comment (crisanlucid) [inline conf/application.conf:383]: what does means "ratingsSortedList" , a new filter ?

File: app/stream/RatingsStream.scala [hunk 2]
```diff
@@ -19,34 +19,28 @@ class RatingsStream @Inject()(ws: WSClient,
...
-  private val maybeUrl = getUrl
+  private val url = getUrl
 
   streamRatingsToRedis()
 
   def streamRatingsToRedis(): Unit = {
+    ws.url(url).withMethod("GET").stream().foreach { response =>
+      val sink = Sink.foreach[ByteString] { bytes =>
+        val maybeRating = decodeRatingMessage(bytes)
+        println(bytes.map(_.toChar).mkString.substring(6))
```
Comment (crisanlucid) [inline app/stream/RatingsStream.scala]: remove debug code

File: app/controllers/RatingsController.scala [hunk 3]
```diff
@@ -11,11 +11,11 @@ import scala.concurrent.{ExecutionContext, Future}
...
 class RatingsController @Inject()(cc: ControllerComponents,
                                   actorSystem: ActorSystem,
-                                  ratingStats: RatingStats)
+                                  ratingStats: RatingStats,
+                                  redisStoreClient: RedisStoreClient)
                                  (implicit exec: ExecutionContext) extends AbstractController(cc) {
   def ratingInfo(movieId: Int): Action[AnyContent] = Action.async { _ =>
-    val redisHandler = new RedisStoreClient
-    val median = redisHandler.getMedian
+    val median = redisStoreClient.getMedian
```
Comment (crisanlucid) [inline app/controllers/RatingsController.scala:18]: can you give me some reference about Median?

File: app/stream/RedisStoreClient.scala [hunk 4]
```diff
@@ -15,12 +16,9 @@ trait IStoreClient {
   def ratingAverage(movieId: Int): RatingAverage
 }
 
-class RedisStoreClient @Inject() extends IStoreClient {
-
-  import com.redis._
+class RedisStoreClient @Inject()(redisClient: RedisClient) extends IStoreClient {
```
Comment (crisanlucid) [inline app/stream/RedisStoreClient.scala]: val ratingsRedisSortedList = "ratingsAvg2"; 

please update the value

Comment (crisanlucid) [inline app/stream/RedisStoreClient.scala]: val ratingsRedisSortedList = "ratingsAvg2"; 

please update the value

File: conf/application.conf (hunk 1 above)
Comment (cricanr) [reply conf/application.conf:383]: "ratingsSortedList" is actually a data structure of type SortedSet in Redis, see: https://redis.io/topics/data-types. Basically what I am doing is I dynamically calculate the average rating of a movie as entries come from the stream and update the value stored in Redis on the fly. 

Redis will create a data structure if none exist with that name automatically, being more like a nosql db.  Let me know if this is enough clarification.

File: app/controllers/RatingsController.scala (hunk 3 above)
Comment (cricanr) [reply app/controllers/RatingsController.scala:18]: Sent email with instructions / requirements and here is a general median description: https://en.wikipedia.org/wiki/Median
Let me know if you want more clarification

**Review by crisanlucid — APPROVED** [review]

**Pull request merged**